[Bug]: Conky 1.18.x fails to compile on Slackware if both Lua 5.1.x and Lua-5.3 are installed.

### What happened?

I maintain the Conky packages for Slackware over at www.slackbuilds.org.  On Slackware Conky requires two dependencies, Lua-5.3 and Imlib2.  However Lua-5.1,5.2, and 5.3 can coexsist if the user wants to install them.  The newer versions of Conky now fail to compile if Lua-5.1.x is installed.  This is an easy workaround, uninstalling 5.1.x allows Conky to compile, but I thought it was worth reporting to see if there is possibly a fix so that uninstalling lua-5.1.x won't be necessary.  

### Version

1.18.x

### Which OS/distro are you seeing the problem on?

Slackware 15.0 / -Current

### Conky config

_No response_

### Stack trace

_No response_

### Relevant log output

```Shell
conky-1.18.0/web/utils/tailwind-preset.js
-- The C compiler identification is GNU 11.2.0
-- The CXX compiler identification is GNU 11.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- CMAKE_C_FLAGS: -O2 -fPIC
-- CMAKE_CXX_FLAGS: -O2 -fPIC
-- CMAKE_C_FLAGS_DEBUG: -g
-- CMAKE_CXX_FLAGS_DEBUG: -g
-- CMAKE_C_FLAGS_MINSIZEREL: -Os -DNDEBUG
-- CMAKE_CXX_FLAGS_MINSIZEREL: -Os -DNDEBUG
-- CMAKE_C_FLAGS_RELEASE: -O3 -DNDEBUG
-- CMAKE_CXX_FLAGS_RELEASE: -O3 -DNDEBUG
-- CMAKE_C_FLAGS_RELWITHDEBINFO: -O2 -g -DNDEBUG
-- CMAKE_CXX_FLAGS_RELWITHDEBINFO: -O2 -g -DNDEBUG
-- CMAKE_BUILD_TYPE: Release
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Looking for include file sys/statfs.h
-- Looking for include file sys/statfs.h - found
-- Looking for include file sys/param.h
-- Looking for include file sys/param.h - found
-- Looking for include file sys/inotify.h
-- Looking for include file sys/inotify.h - found
-- Looking for include file dirent.h
-- Looking for include file dirent.h - found
-- Looking for strndup
-- Looking for strndup - found
-- Looking for pipe2
-- Looking for pipe2 - not found
-- Looking for O_CLOEXEC
-- Looking for O_CLOEXEC - found
-- Looking for statfs64
-- Looking for statfs64 - found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for include file linux/soundcard.h
-- Looking for include file linux/soundcard.h - found
-- Looking for include file linux/soundcard.h
-- Looking for include file linux/soundcard.h - found
-- Looking for include file linux/sockios.h
-- Looking for include file linux/sockios.h - found
-- Looking for wsyncup in /usr/lib64/libcurses.so
-- Looking for wsyncup in /usr/lib64/libcurses.so - found
-- Looking for cbreak in /usr/lib64/libncurses.so
-- Looking for cbreak in /usr/lib64/libncurses.so - found
-- Looking for nodelay in /usr/lib64/libncurses.so
-- Looking for nodelay in /usr/lib64/libncurses.so - found
-- Found Curses: /usr/lib64/libncurses.so  
-- Looking for include file iwlib.h
-- Looking for include file iwlib.h - found
-- Looking for iw_sockets_open
-- Looking for iw_sockets_open - not found
-- Looking for getnameinfo
-- Looking for getnameinfo - found
-- Looking for 5 include files netdb.h, ..., arpa/inet.h
-- Looking for 5 include files netdb.h, ..., arpa/inet.h - found
-- Found X11: /usr/include   
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found Lua: /usr/lib64/liblua.so;/usr/lib64/libm.so (found version "5.1.5") 
-- Checking for modules 'cairo>=1.14;cairo-xlib'
--   Found cairo, version 1.16.0
--   Found cairo-xlib, version 1.16.0
-- Checking for one of the modules 'imlib2;Imlib2'
-- Checking for module 'libpulse'
--   Found libpulse, version 15.0
-- Checking for module 'libcurl'
--   Found libcurl, version 7.88.1
-- Found LibXml2: /usr/lib64/libxml2.so (found version "2.9.14") 
-- Found Python3: /usr/bin/python3.9 (found version "3.9.16") found components: Interpreter 
Configuration for target conky-cairo
-- SOURCE_DIR = /tmp/SBo/conky-1.18.0/lua
-- BINARY_DIR = /tmp/SBo/conky-1.18.0/build/lua
-- COMPILE_OPTIONS = $<$<COMPILE_LANG_AND_ID:CXX,Clang>:-stdlib=libc++>
-- INCLUDE_DIRECTORIES = /tmp/SBo/conky-1.18.0/3rdparty/toluapp/include;/usr/include/cairo;/usr/include/pixman-1;/usr/include/libxml2;/usr/include/freetype2;/usr/include/harfbuzz;/usr/include/glib-2.0;/usr/lib64/glib-2.0/include;/usr/include/libpng16;/usr/include;/tmp/SBo/conky-1.18.0/lua
-- LINK_LIBRARIES = cairo;X11;Xext;/usr/lib64/liblua.so;/usr/lib64/libm.so;toluapp_lib_static
Configuration for target conky-imlib2
-- SOURCE_DIR = /tmp/SBo/conky-1.18.0/lua
-- BINARY_DIR = /tmp/SBo/conky-1.18.0/build/lua
-- COMPILE_OPTIONS = $<$<COMPILE_LANG_AND_ID:CXX,Clang>:-stdlib=libc++>
-- INCLUDE_DIRECTORIES = /tmp/SBo/conky-1.18.0/3rdparty/toluapp/include;/usr/include/cairo;/usr/include/pixman-1;/usr/include/libxml2;/usr/include/freetype2;/usr/include/harfbuzz;/usr/include/glib-2.0;/usr/lib64/glib-2.0/include;/usr/include/libpng16;/usr/include;/tmp/SBo/conky-1.18.0/lua;/usr/include;/usr/include;/tmp/SBo/conky-1.18.0/lua
-- LINK_LIBRARIES = -lImlib2;/usr/lib64/liblua.so;/usr/lib64/libm.so;toluapp_lib_static
Configuration for target conky
-- SOURCE_DIR = /tmp/SBo/conky-1.18.0/src
-- BINARY_DIR = /tmp/SBo/conky-1.18.0/build/src
-- COMPILE_OPTIONS = $<$<COMPILE_LANG_AND_ID:CXX,Clang>:-stdlib=libc++>
-- INCLUDE_DIRECTORIES = /tmp/SBo/conky-1.18.0/3rdparty/toluapp/include;/tmp/SBo/conky-1.18.0/build;/usr/include;/usr/include;/usr/include/freetype2;/usr/include;/usr/include;/usr/include/libxml2;/tmp/SBo/conky-1.18.0/build/data
-- LINK_LIBRARIES = conky_core;-lpthread;-lm;/usr/lib64/libncurses.so;/usr/lib64/libform.so;/usr/lib64/libiw.a;/usr/lib64/libSM.so;/usr/lib64/libICE.so;/usr/lib64/libX11.so;/usr/lib64/libXext.so;/usr/lib64/libXdamage.so;/usr/lib64/libXfixes.so;/usr/lib64/libXft.so;/usr/lib64/libXext.so;/usr/lib64/libXinerama.so;/usr/lib64/libXfixes.so;/usr/lib64/liblua.so;/usr/lib64/libm.so;/usr/lib64/libXNVCtrl.so;-lImlib2;pulse;curl;curl;/usr/lib64/libxml2.so;toluapp_lib_static;tcp-portmon
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/SBo/conky-1.18.0/build
[  1%] Building CXX object src/CMakeFiles/tcp-portmon.dir/libtcp-portmon.cc.o
[  2%] Building C object 3rdparty/toluapp/CMakeFiles/toluapp_lib_static.dir/src/lib/tolua_event.c.o
[  3%] Generating defconfig.h
[  4%] Generating convertconf.h
[  5%] Building C object 3rdparty/toluapp/CMakeFiles/toluapp_lib_static.dir/src/lib/tolua_map.c.o
[  6%] Building C object 3rdparty/toluapp/CMakeFiles/toluapp_lib_static.dir/src/lib/tolua_push.c.o
[  8%] Building C object 3rdparty/toluapp/CMakeFiles/toluapp_lib_static.dir/src/lib/tolua_to.c.o
[  8%] Building C object 3rdparty/toluapp/CMakeFiles/toluapp_lib_static.dir/src/lib/tolua_is.c.o
[  8%] Built target generated_hdr_files
[  9%] Linking C static library libtoluapp.a
[  9%] Built target toluapp_lib_static
[ 10%] Building C object 3rdparty/toluapp/CMakeFiles/toluapp.dir/src/bin/tolua.c.o
[ 11%] Building C object 3rdparty/toluapp/CMakeFiles/toluapp.dir/src/bin/toluabind.c.o
[ 12%] Linking C executable toluapp
[ 12%] Built target toluapp
[ 13%] Running tolua++ on imlib2.pkg
[ 14%] Running tolua++ on cairo_imlib2_helper.pkg
[ 15%] Running tolua++ on cairo.pkg

** tolua internal error: [string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:337: bad argument #1 to 'load' (function expected, got string)
stack traceback:
	[C]: in function 'load'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:337: in function 'prep'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:253: in function 'Package'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:54: in function <[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:27>
	[C]: in function 'xpcall'
	...Bo/conky-1.18.0/3rdparty/toluapp/src/bin/lua/all.lua:25: in main chunk.



** tolua internal error: [string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:337: bad argument #1 to 'load' (function expected, got string)
stack traceback:
	[C]: in function 'load'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:337: in function 'prep'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:253: in function 'Package'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:54: in function <[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:27>
	[C]: in function 'xpcall'
	...Bo/conky-1.18.0/3rdparty/toluapp/src/bin/lua/all.lua:25: in main chunk.


[ 15%] Built target conkyrc.vim

** tolua internal error: [string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:337: bad argument #1 to 'load' (function expected, got string)
stack traceback:
	[C]: in function 'load'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:337: in function 'prep'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:253: in function 'Package'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:54: in function <[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:27>
	[C]: in function 'xpcall'
	...Bo/conky-1.18.0/3rdparty/toluapp/src/bin/lua/all.lua:25: in main chunk.


[ 15%] Built target conky.nanorc
[ 15%] Running tolua++ on cairo_imlib2_helper.pkg
[ 16%] Patching libcairo-orig.c
[ 16%] Running tolua++ on imlib2.pkg
1 out of 1 hunk FAILED -- saving rejects to file /tmp/SBo/conky-1.18.0/build/lua/libcairo.c.rej
make[2]: *** [lua/CMakeFiles/conky-cairo.dir/build.make:75: lua/libcairo.c] Error 1
make[2]: *** Deleting file 'lua/libcairo.c'
make[1]: *** [CMakeFiles/Makefile2:327: lua/CMakeFiles/conky-cairo.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

** tolua internal error: [string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:337: bad argument #1 to 'load' (function expected, got string)
stack traceback:
	[C]: in function 'load'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:337: in function 'prep'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:253: in function 'Package'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:54: in function <[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:27>
	[C]: in function 'xpcall'
	...Bo/conky-1.18.0/3rdparty/toluapp/src/bin/lua/all.lua:25: in main chunk.


[ 17%] Building C object lua/CMakeFiles/conky-cairo_imlib2_helper.dir/libcairo_imlib2_helper.c.o
cc1: fatal error: /tmp/SBo/conky-1.18.0/build/lua/libcairo_imlib2_helper.c: No such file or directory
compilation terminated.
make[2]: *** [lua/CMakeFiles/conky-cairo_imlib2_helper.dir/build.make:80: lua/CMakeFiles/conky-cairo_imlib2_helper.dir/libcairo_imlib2_helper.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:274: lua/CMakeFiles/conky-cairo_imlib2_helper.dir/all] Error 2

** tolua internal error: [string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:337: bad argument #1 to 'load' (function expected, got string)
stack traceback:
	[C]: in function 'load'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:337: in function 'prep'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:253: in function 'Package'
	[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:54: in function <[string "/tmp/SBo/conky-1.18.0/3rdparty/toluapp/src/..."]:27>
	[C]: in function 'xpcall'
	...Bo/conky-1.18.0/3rdparty/toluapp/src/bin/lua/all.lua:25: in main chunk.


[ 18%] Building C object lua/CMakeFiles/conky-imlib2.dir/libimlib2.c.o
cc1: fatal error: /tmp/SBo/conky-1.18.0/build/lua/libimlib2.c: No such file or directory
compilation terminated.
make[2]: *** [lua/CMakeFiles/conky-imlib2.dir/build.make:80: lua/CMakeFiles/conky-imlib2.dir/libimlib2.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:301: lua/CMakeFiles/conky-imlib2.dir/all] Error 2
[ 19%] Linking CXX static library libtcp-portmon.a
[ 19%] Built target tcp-portmon
make: *** [Makefile:166: all] Error 2
```

--- a/cmake/ConkyPlatformChecks.cmake
+++ b/cmake/ConkyPlatformChecks.cmake
@@ -442,7 +442,7 @@ if(BUILD_WAYLAND)
   set(conky_includes ${conky_includes} ${PANGOFT2_INCLUDE_DIRS})
 endif(BUILD_WAYLAND)
 
-include(FindLua)
+find_package(Lua "5.3" REQUIRED)
 
 set(conky_libs ${conky_libs} ${LUA_LIBRARIES})
 set(conky_includes ${conky_includes} ${LUA_INCLUDE_DIR})
@@ -465,9 +465,9 @@ if(BUILD_X11)
     pkg_search_module(IMLIB2 REQUIRED imlib2 Imlib2)
     set(luaimlib2_libs ${IMLIB2_LIBS} ${IMLIB2_LDFLAGS} ${LUA_LIBRARIES})
     set(luaimlib2_includes
-        ${IMLIB2_INCLUDE_DIRS}
-        ${LUA_INCLUDE_DIR}
-        ${X11_INCLUDE_DIR})
+      ${IMLIB2_INCLUDE_DIRS}
+      ${LUA_INCLUDE_DIR}
+      ${X11_INCLUDE_DIR})
   endif(BUILD_LUA_IMLIB2)
 
   if(BUILD_LUA_RSVG)

